include/modem/trace_medium.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/**@file trace_medium.h
+ *
+ * @defgroup trace_medium nRF91 Modem trace medium interface
+ * @{
+ */
+#ifndef TRACE_MEDIUM_H__
+#define TRACE_MEDIUM_H__
+
+#include <stdbool.h>
+#include <stdint.h>
+
+/**
+ * @brief Initialize the compiled trace medium.
+ *
+ * @retval Zero on success.
+ * @retval Negative error code on failure.
+ */
+int trace_medium_init(void);
+
+/**
+ * @brief Deinitialize the compiled trace medium.
+ *
+ * @retval Zero on success.
+ * @retval Negative error code on failure.
+ */
+int trace_medium_deinit(void);
+
+/**
+ * @brief Send trace data to the compiled trace medium.
+ *
+ * @param data Memory buffer containing the modem trace data.
+ * @param len  Memory buffer length.
+ */
+void trace_medium_write(const uint8_t *data, uint32_t len);
```

Comment:
I think `_(const void *data, size_t len)` would be a better signature. It's better to cast the pointer inside this function, if we need to, than let the caller do the cast, also `size_t` is better to express the size of an object in memory.